specs/Print.md

```diff
@@ -0,0 +1,816 @@
+// COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER, opens a browser print preview dialog,
+// otherwise opens a system print dialog.
+
+bool AppWindow::ShowPrintUI(COREWEBVIEW2_PRINT_DIALOG_KIND printDialogKind)
+{
+  wil::com_ptr<ICoreWebView2_15> webView2_15;
+  CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+  CHECK_FEATURE_RETURN(webView2_15);
+  CHECK_FAILURE(webView2_15->ShowPrintUI(printDialogKind));
+  return true;
+}
+```
+
+```c#
+void ShowPrintUI(object target, ExecutedRoutedEventArgs e)
+{
+  string printDialog = e.Parameter.ToString();
+  if (printDialog == "Browser")
+  {
+    // Opens the browser print preview dialog.
+    webView.CoreWebView2.ShowPrintUI();
+  }
+  else
+  {
+    // Opens the system print dialog.
+    webView.CoreWebView2.ShowPrintUI(CoreWebView2PrintDialogKind.System);
+  }
+}
+```
+
+# Print to default printer
+
+You can use `Print` method to print to a default printer without a print dialog.
+You can either use default page and printer settings or programmatically configure
+print settings, and call `Print` to print the current web page.
+
+```cpp
+// This example prints the current web page without a print dialog to a default printer
+// with the default print settings.
+
+bool AppWindow::PrintToDefaultPrinter()
+{
+  wil::com_ptr<ICoreWebView2_15> webView2_15;
+  CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+
+  wil::com_ptr<ICoreWebView2Environment6> webviewEnvironment6;
+  CHECK_FAILURE(m_webViewEnvironment->QueryInterface(IID_PPV_ARGS(&webviewEnvironment6)));
+
+  wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+  CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+  wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+  CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  // Prints current web page with the default page and printer settings.
+  CHECK_FAILURE(webView2_15->Print(
+      printSettings2.get(),
+      Callback<ICoreWebView2PrintCompletedHandler>(
+          [title = title.get(), this](HRESULT errorCode, BOOL isSuccessful) -> HRESULT
+          {
+            std::wstring message = L"";
+            if (errorCode == S_OK && isSuccessful)
+            {
+              message = L"Printing " + std::wstring(title) +
+                              L" document to printer is succedded";
+            }
+            else
+            {
+              if (errorCode == S_OK)
+              {
+                message = L"Printer is not available, offline or error state";
+              }
+              else if (errorCode == E_ABORT)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document already in progress";
+              }
+              else if (errorCode == E_FAIL)
```

Comment:
```suggestion
              else
```
Presumably apps should treat any other error as a generic failure, rather than falling through with an empty message.

Reply:
Will do

Reply:
Please fix.